protobuf/module_msg.proto

```diff
@@ -6,385 +6,869 @@ package bess.pb;
+    string filter = 2; /// The actual BPF string.
+    int64 gate = 3; ///What gate to forward packets that match this BPF to.
   }
-  repeated Filter filters = 1;
+  repeated Filter filters = 1; /// The BPF initialized function takes a list of BPF filters.
 }
 
+/**
+ * The Buffer module takes no parameters to initialize (ie, `Buffer()` is sufficient to create one).
+ * Buffer accepts packets and stores them; it may forard them to the next module only after it has
+ * received enough packets to fill an entire PacketBatch.
+ *
+ * __Input Gates__: 1
+ * __Output Gates__: 1
+ */
 message BufferArg {
 }
 
+/**
+ * The Bypass module forwards packets without any processing. It requires no parameters to initialize. Bypass is useful primarily for testing and performance evaluation.
+ *
+ * __Input Gates__: 1
+ * __Output Gates__: 1
+ */
 message BypassArg {
 }
 
+/**
+ * The Dump module blindly forwards packets without modifying them. It periodically samples a packet and prints out out to the BESS log (by default stored in `/tmp/bessd.INFO`).
+ *
+ * __Input Gates__: 1
+ * __Output Gates__: 1
+ */
 message DumpArg {
-  double interval = 1;
+  double interval = 1; ///How frequently to sample and print a packet, in seconds.
 }
 
+/**
+ * The EtherEncap module wraps packets in an Ethernet header, but it takes no parameters. Instead, Ethernet source, destination, and type are pulled from a packet's metadata attributes.
+ * For example: `SetMetadata('dst_mac', 11:22:33:44:55) -> EtherEncap()`
+ * This is useful when upstream modules wish to assign a MAC address to a packet, e.g., due to an ARP request.
+ *
+ * __Input Gates__: 1
+ * __Output Gates__: 1
+ */
 message EtherEncapArg {
 }
 
+/**
+ * The ExactMatch module splits packets along output gates according to exact match values in arbitrary packet fields.
+ * To instantiate an ExactMatch module, you must specify which fields in the packet to match over. You can add rules using the function `ExactMatch.add(...)`
+ * Fields may be stored either in the packet data or its metadata attributes.
+ * An example script using the ExactMatch code is found
+ * in [`bess/bessctl/conf/samples/exactmatch.bess`](https://github.com/NetSys/bess/blob/master/bessctl/conf/samples/exactmatch.bess).
+ *
+ * __Input Gates__: 1
+ * __Output Gates__: many (configurable)
+ */
 message ExactMatchArg {
+  /**
+   * An ExactMatch Field specifies a field over which to check for ExactMatch rules. Field may be in EITHER the packet's data OR it's metadata attributes.
+   */
   message Field {
-    uint64 size = 1;
-    uint64 mask = 4;
+    uint64 size = 1; /// The length, in bytes, of the field to inspect.
+    uint64 mask = 4; /// A bitmask over the field to specify which bits to inspect (default 0xff).
```

Comment:
Is this the decimal representation of the mask?  If so, what byte ordering is it in (I assume host byte order)?  Also, does the fact that the mask is a uint64 mean that the size field above is implicitly limited to 64 bits?